Jenkinsfile

```diff
@@ -55,120 +62,103 @@ pipeline {
         }
         stage('Build') {
           steps {
+            sh "git clone -b develop git@github.com:xmos/xcommon_cmake ${WORKSPACE}/xcommon_cmake"
             dir('xscope_fileio') {
               withTools(params.TOOLS_VERSION) {
-                sh 'tree'
-                sh 'cd examples/throughput_c && make'
-                sh 'cd tests/no_hang && . make.sh'
-                withEnv(["XMOS_MODULE_PATH=${WORKSPACE}", "XCOMMON_DISABLE_AUTO_MODULE_SEARCH=1"]) {
-                  sh 'cd examples/fileio_features_xc && xmake'
-                }
-                // xcommon cmake
-                sh "git clone -b develop git@github.com:xmos/xcommon_cmake ${WORKSPACE}/xcommon_cmake"
                 withEnv(["XMOS_CMAKE_PATH=${WORKSPACE}/xcommon_cmake"]) {
-                  // build close files test
-                  sh 'cmake -G "Unix Makefiles" -S tests/close_files -B tests/close_files/build'
-                  sh 'xmake -C tests/close_files/build -j4'
+                  buildApps([
+                    "examples/fileio_features_xc",
+                    "examples/throughput_c",
+                    "tests/no_hang",
+                    "tests/close_files",
+                  ]) // buildApps
                 } // withEnv
               } // withTools
             } // dir
           } // steps
         } // stage 'Build'
-
         stage('Cleanup xtagctl'){
           steps {
             dir('xscope_fileio') {
               withVenv() {
                 withTools(params.TOOLS_VERSION) {
-                  // sh 'rm -f ~/.xtag/status.lock ~/.xtag/acquired' // not needed
+                  sh 'rm -f ~/.xtag/status.lock ~/.xtag/acquired' // not needed
                   sh 'xtagctl reset_all XCORE-AI-EXPLORER'
                 }
               }
             }
           }
         }
-        
         stage('Tests'){
           failFast false
           parallel {
-            stage('Hardware tests') {
+
+            stage('Hardware tests #1 (in parallel)') {
               stages{
                 stage('Transfer test single large'){
                   steps {
-                    dir('xscope_fileio') {
+                    dir('xscope_fileio/tests') {
                       withVenv() {
                         withTools(params.TOOLS_VERSION) {
-                          sh 'python tests/test_throughput.py 64' //Pass size in MB
-                        }
-                      }
-                    }
-                  }
-                }
-              }
-            }
+                          sh 'pytest test_throughput.py \
+                              -k test_throughput_1 \
```

Comment:
Splitting the testcases by keyword across separate parallel stages feels fragile: if someone adds a new testcase in test_throughput.py and just runs the job, that testcase will never be selected and the job could succeed, but the testcase could be failing.

It's also unclear to me why these tests are done in parallel: the total duration is less than three minutes - why not go for simplicity and just run it all in a single stage?